server/topic.go

```diff
@@ -413,23 +413,50 @@ func (t *Topic) handleCtrlBroadcast(msg *ServerComMessage) {
 	}
 }
 
+// updateAcsFromPresMsg modifies user acs in Topic's perUser struct based on the data in `pres`.
 func (t *Topic) updateAcsFromPresMsg(pres *MsgServerPres) {
 	uid := types.ParseUserId(pres.Src)
 	dacs := pres.Acs
-	if pud, ok := t.perUser[uid]; !uid.IsZero() && ok {
+	if uid.IsZero() {
+		log.Printf("proxy topic[%s]: received acs change for invalid user id '%s'", t.name, pres.Src)
+		return
+	}
+	if pud, ok := t.perUser[uid]; ok {
+		// Existing user. Update.
 		if dacs.Want != "" {
```

Comment:
This check is unnecessary because `ApplyDelta` checks it anyway.

Reply:
Done.